component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -134,18 +146,50 @@ public AuthenticatorFlowStatus process(HttpServletRequest request,
             return AuthenticatorFlowStatus.INCOMPLETE;
         } else if (StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.CODE)) &&
                 StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.RESEND))) {
-            // if the request comes with code, it will go through this flow.
-            initiateAuthenticationRequest(request, response, context);
-            publishPostEmailOTPGeneratedEvent(request, context);
-            if (context.getProperty(EmailOTPAuthenticatorConstants.AUTHENTICATION)
-                    .equals(EmailOTPAuthenticatorConstants.AUTHENTICATOR_NAME)) {
-                // if the request comes with authentication is EmailOTP, it will go through this flow.
-                // set the current authenticator name
-                context.setCurrentAuthenticator(getName());
-                return AuthenticatorFlowStatus.INCOMPLETE;
+            User user;
+            AuthenticatedUser authenticatedUser = getAuthenticatedUser(context);
+            if (authenticatedUser == null) {
+                if (StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.USER_NAME))) {
+                    redirectUserToIDF(response, context);
+                    context.setProperty(IS_IDF_INITIATED_FROM_AUTHENTICATOR, true);
```

Comment:
Let's have the class qualifier. It gives a better context as to where a constant is imported from
```suggestion
                    context.setProperty(EmailOTPAuthenticatorConstants.IS_IDF_INITIATED_FROM_AUTHENTICATOR, true);
```

Reply:
Fixed